### PR TITLE
New article - data navigation - 3

### PR DESCRIPTION
**Pull request by rachana-satao**
This article is merging of [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/how-to-navigate-data-in-windows-forms?view=netframeworkdesktop-4.8), [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/ensure-the-selected-row-in-a-child-table-correct?view=netframeworkdesktop-4.8)

Preview - [link](https://review.docs.microsoft.com/en-us/dotnet/desktop/winforms/data/work-with-data-grid?view=netdesktop-5.0&branch=pr-en-us-1384)

### CODE REVIEW

**Comment by rachana-satao** [comment]
@adegeo Acrolinx score for this article - 98

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+## To increment the position in a data source
+
+01. Set the <xref:System.Windows.Forms.BindingSource.Position%2A> property of the `BindingSource` for your bound data to the record position to go to. The following example illustrates using the <xref:System.Windows.Forms.BindingSource.MoveNext%2A> method of the `BindingSource` to increment the `Position` property when the `nextButton` is clicked. The `BindingSource` is associated with the `Customers` table of a dataset `Northwind`.
+
+    > [!NOTE]
+    > Setting the <xref:System.Windows.Forms.BindingSource.Position%2A> property to a value beyond the first or last record does not result in an error, as the .NET will not allow you to set the position to a value outside the bounds of the list. If it's important in your application to know whether you have gone past the first or last record, include logic to test whether you will exceed the data element count.
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:26]: .NET won't let you>> use contraction here and change allow you to let you

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form1.cs" id="find_the_record":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form1.vb" id="find_the_record":::
+
+## How to ensure the selected row in a child table remains at the correct position
+
+When you work with data binding in Windows Forms, you'll display data in a parent/child or master/details view. It's a data-binding scenario where data from the same source is displayed in two controls. Changing the selection in one control causes the data displayed in the second control to change. For example, the first control might contain a list of customers and the second a list of orders related to the selected customer in the first control.
+
+When you display data in a parent/child view, you might have to take extra steps to ensure that the currently selected row in the child table isn't reset to the first row of the table. In order to do this, you'll have to cache the child table position and reset it after the parent table changes. Typically the child reset occurs the first time a field in a row of the parent table changes.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:54]: Though it's understood for more clarity we can modify the sentence as:
**Typically,** the child **table** reset occurs the first time a field in a row of the parent table changes.

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+### To cache the current child position
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:56]: To cache the current child table's position

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+01. Declare an integer variable to store the child list position and a Boolean variable to store whether to cache the child position.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:58]: It's better to say "child table"

Declare an integer variable to store the child **table's** list position and a Boolean variable to store whether to cache the child **table's** position.

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="store_the_child_list_pos":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="store_the_child_list_pos":::
+
+01. Handle the <xref:System.Windows.Forms.CurrencyManager.ListChanged> event for the binding's <xref:System.Windows.Forms.CurrencyManager> and check for a <xref:System.ComponentModel.ListChangedType> of <xref:System.ComponentModel.ListChangedType.Reset>.
+
+01. Check the current position of the <xref:System.Windows.Forms.CurrencyManager>. If it's greater than first entry in the list (typically 0), save it to a variable.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:65]: If it's greater than **the** first entry in the list (typically 0), save it to a variable.

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="check_current_pos_of_currency_manager":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="check_current_pos_of_currency_manager":::
+
+01. Handle the parent list's <xref:System.Windows.Forms.BindingManagerBase.CurrentChanged> event for the parent currency manager. In the handler, set the Boolean value to indicate it isn't a caching scenario. If the `CurrentChanged` occurs, the change to the parent is a list position change and not an item value change.
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="handle_current_change_event":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="handle_current_change_event":::
+
+### To reset the child position
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:75]: To reset the child table's position

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+01. Handle the <xref:System.Windows.Forms.BindingManagerBase.PositionChanged> event for the child binding's <xref:System.Windows.Forms.CurrencyManager>.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:77]: Handle the <xref:System.Windows.Forms.BindingManagerBase.PositionChanged> event for the child **table** binding's....

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+01. Reset the child table position to the cached position saved in the previous procedure.
+
+     :::code language="csharp" source="snippets/work-with-data-grid/csharp/form2.cs" id="reset_child_tab_pos":::
+     :::code language="vb" source="snippets/work-with-data-grid/vb/form2.vb" id="reset_child_tab_pos":::
+
+To test the code example, perform the following steps:
+
+01. Run the example.
+
+01. Ensure the **Cache and reset position** check box is selected.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:88]: **checkbox** is a single word. Use active. The rest of the steps are in active voice.

Modify the sentence as "Select the Cache and reset position checkbox."

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+01. Select the **Clear parent field** button to cause a change in a field of the parent table. Notice that the selected row in the child table doesn't change.
+
+01. Close and run the example again. You need to run it again because the reset behavior occurs only on the first change in the parent row.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:92]: Close and **rerun the example**. You need to run it again.....

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+01. Clear the **Cache and reset position** check box.
```
Comment (SoumyaRamani) [inline dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:94]: checkbox

**Comment by SoumyaRamani** [comment]
@rachana-satao  

I have reviewed the article. Please check and let me know if you have any questions.

cc: @Lovina-Saldanha

File: dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  - "data sources [Windows Forms], Windows Forms"
+  - "Windows Forms, navigating"
+  - "CurrencyManager class [Windows Forms], navigating Windows Forms data"
+  - "data [Windows Forms], navigating"
+ms.custom: devdivchpfy22
+---
+
+# How to navigate data in Windows Forms .NET
+
+In a Windows application, the easiest way to navigate through records in a data source is to bind a <xref:System.Windows.Forms.BindingSource> component to the data source and then bind controls to the `BindingSource`. You can then use the built-in navigation method on the `BindingSource` such a <xref:System.Windows.Forms.BindingSource.MoveNext%2A>, <xref:System.Windows.Forms.BindingSource.MoveLast%2A>, <xref:System.Windows.Forms.BindingSource.MovePrevious%2A> and <xref:System.Windows.Forms.BindingSource.MoveFirst%2A>. Using these methods will adjust the <xref:System.Windows.Forms.BindingSource.Position%2A> and <xref:System.Windows.Forms.BindingSource.Current%2A> properties of the `BindingSource` appropriately. You can also find an item and set it as the current item by setting the `Position` property.
```
[the remark this answers is not in the document]
Comment (rachana-satao) [reply dotnet-desktop-guide/net/winforms/data/work-with-data-grid.md:19]: ok

**Review by adegeo — APPROVED** [review]

**Pull request merged**